File: leetcode/C++/detect-capital.cpp
class Solution {
public:
    bool detectCapitalUse(string word) {
        int upperLen = 0;
        for(char &ch : word) {if(ch>='A' && ch<='Z') ++upperLen;}
        return (upperLen == word.length()) || (upperLen==1 && word[0]>='A' && word[0]<='Z') || (upperLen == 0);
    }
};

File: leetcode/Python/detect-capital.py
class Solution:
    def detectCapitalUse(self, word: str) -> bool:
        upperLen = 0
        for w in word:
            if 'A'<=w<='Z':
                upperLen += 1
        return (upperLen == len(word)) or (upperLen == 0) or (upperLen == 1 and 'A'<=word[0]<='Z')